pkg/acme/client/middleware/logger.go

```diff
@@ -108,5 +157,24 @@ func (l *Logger) Discover(ctx context.Context) (acme.Directory, error) {
 
 func (l *Logger) UpdateReg(ctx context.Context, a *acme.Account) (*acme.Account, error) {
 	klog.Infof("Calling UpdateAccount")
-	return l.baseCl.UpdateReg(ctx, a)
+
+	acmectx, cancel := l.acmeContext(ctx)
+	defer cancel()
+
+	return l.baseCl.UpdateReg(acmectx, a)
+}
+
+func (l *Logger) acmeContext(ctx context.Context) (context.Context, context.CancelFunc) {
+	acmectx, cancel := context.WithTimeout(context.Background(), timeout)
+
+	go func() {
+		select {
+		case <-ctx.Done():
+			cancel()
+		case <-acmectx.Done():
+			return
+		}
+	}()
```

Comment:
This isn't necessary - contexts are already 'layered', i.e. if the top level context is cancelled, all child contexts are automatically cancelled too.

Reply:
Not sure what you mean by this? acmecxt isn't a child of ctx no? Unless I'm miss understanding.

Surely we need this to cancel the acme client context if the controller context is cancelled?